crnk-example-frontend/src/app/common/language/common.language.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { Store } from '@ngrx/store';
+import { SetLanguageAction } from '../../store';
+
+@Injectable()
+export class AppLanguageService {
+
+	langs = [LanguageCode.en, LanguageCode.de];
+	defaultLang = LanguageCode.en;
+
+	constructor(
+		private store: Store<any>,
+		private translate: TranslateService
+	) {
+		this.translate.addLangs(this.langs);
+		this.translate.setDefaultLang(this.defaultLang);
+
+		this.changeLanguage(this.getLanguageFromBrowser());
+	}
+
+	private getLanguageFromBrowser(): LanguageCode {
+		const language = localStorage.getItem('language');
+		if (language) {
+			return language as LanguageCode;
+		}
+		else {
+			const browserLang = this.translate.getBrowserLang();
+			const languageToSelect = this.langs.includes(browserLang as LanguageCode) ? browserLang : this.defaultLang;
+			return languageToSelect as LanguageCode;
+		}
+	}
+
+	changeLanguage(language: LanguageCode) {
+		localStorage.setItem('language', language);
+		this.translate.use(language);
```

Comment:
those two elements (setting local storage and updating translate could go into an ngrx effect? then anybody can just trigger the setLanguageAction, but it does only set the store.

Reply:
getLanguageFromBrowser could potentially be put into an APP_INITIALIZER and then trigger the setLanguage action on startup? or maybe the effect initializes the defaults like here.